cloud/deployment/fromfile/fromfile.go

```diff
@@ -158,189 +148,416 @@ func CreateOrUpdate(inputFile, action string, client astro.Client, coreClient as
-				return astro.CreateDeploymentInput{}, astro.UpdateDeploymentInput{},
-					fmt.Errorf("%s %w more than one worker queue. (%d) were requested",
-						deployment.KubeExecutor, workerqueue.ErrNotSupported, len(listQueues))
-			}
-			for i := range listQueues {
-				err = workerqueue.IsKubernetesWorkerQueueInputValid(&listQueues[i])
+				listQueuesRequest = append(listQueuesRequest, *a)
+			} else {
+				var workerQueue astroplatformcore.HybridWorkerQueueRequest
+				workerQueue.Id = &listQueues[i].Id
+				workerQueue.Name = listQueues[i].Name
+				workerQueue.IsDefault = listQueues[i].IsDefault
+				workerQueue.MaxWorkerCount = listQueues[i].MaxWorkerCount
+				workerQueue.MinWorkerCount = listQueues[i].MinWorkerCount
+				workerQueue.WorkerConcurrency = listQueues[i].WorkerConcurrency
+				workerQueue.NodePoolId = *listQueues[i].NodePoolId
+				// check if queue is valid
+				if deploymentFromFile.Deployment.Configuration.Executor == deployment.KubeExecutor || deploymentFromFile.Deployment.Configuration.Executor == deployment.KUBERNETES {
+					return errUseDefaultWokerType
+				}
+				// set default values if none were specified
+				a := workerqueue.SetWorkerQueueValuesHybrid(listQueues[i].MinWorkerCount, listQueues[i].MaxWorkerCount, listQueues[i].WorkerConcurrency, &workerQueue, defaultOptions)
+				err = workerqueue.IsCeleryWorkerQueueInputValid(a, defaultOptions)
 				if err != nil {
-					return astro.CreateDeploymentInput{}, astro.UpdateDeploymentInput{}, err
+					return err
 				}
+				workerQueue = *a
+				// add it to the list of queues to be created
+				listHybridQueuesRequest = append(listHybridQueuesRequest, workerQueue)
 			}
 		}
 	}
 	switch action {
 	case createAction:
-		createInput = astro.CreateDeploymentInput{
-			WorkspaceID:           workspaceID,
-			ClusterID:             clusterID,
-			Label:                 deploymentFromFile.Deployment.Configuration.Name,
-			Description:           deploymentFromFile.Deployment.Configuration.Description,
-			RuntimeReleaseVersion: deploymentFromFile.Deployment.Configuration.RunTimeVersion,
-			DagDeployEnabled:      dagDeploy,
-			SchedulerSize:         deploymentFromFile.Deployment.Configuration.SchedulerSize,
-			APIKeyOnlyDeployments: deploymentFromFile.Deployment.Configuration.APIKeyOnlyDeployments,
-			IsHighAvailability:    deploymentFromFile.Deployment.Configuration.IsHighAvailability,
-			DeploymentSpec: astro.DeploymentCreateSpec{
-				Executor: deploymentFromFile.Deployment.Configuration.Executor,
-				Scheduler: astro.Scheduler{
-					AU:       deploymentFromFile.Deployment.Configuration.SchedulerAU,
+		createDeploymentRequest := astroplatformcore.CreateDeploymentRequest{}
+		if deployment.IsDeploymentStandard(deploymentType) {
+			var requestedCloudProvider astroplatformcore.CreateStandardDeploymentRequestCloudProvider
+			if deploymentFromFile.Deployment.Configuration.CloudProvider == "gcp" || deploymentFromFile.Deployment.Configuration.CloudProvider == "GCP" {
```

Comment:
nit: we can make this code more readable by doing something like this
```suggestion
			if strings.ToUpper(deploymentFromFile.Deployment.Configuration.CloudProvider) == "GCP" {
```

Reply:
also I think we should use else ifs here
it makes the code blocks more separated and its easier to read

Reply:
still working on this change

Reply:
fixed